### PR TITLE
chore: split knownroles from store [WEB-589]

### PR DESCRIPTION
**Pull request by gt2345**
## Description

As part of breaking parts out of the central Store, this covers the knownRoles.

## Test Plan

Enable rbac, navigate to user management page, select a user to edit, should be able to add/remove roles.
Navigate to group management page, select a user to edit, should be able to add/remove roles.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: webui/react/src/hooks/useModal/UserSettings/useModalCreateGroup.tsx [hunk 1]
```diff
@@ -125,14 +127,18 @@ const ModalForm: React.FC<Props> = ({ form, users, group, groupRoles }) => {
...
+                Loaded: () =>
+                  knownRoles.map((r) => (
+                    <Select.Option
+                      disabled={groupRoles?.find((gr) => gr.id === r.id)?.fromWorkspace?.length}
+                      key={r.id}
+                      value={r.id}>
+                      {r.name}
+                    </Select.Option>
+                  )),
+                NotLoaded: () => undefined,
```
Comment (keita-determined) [inline webui/react/src/hooks/useModal/UserSettings/useModalCreateGroup.tsx:140]: https://github.com/determined-ai/determined/pull/5522#discussion_r1042317154

We should add loading state (spinner) or `// TODO show spinner when this is loading`

File: webui/react/src/hooks/useModal/UserSettings/useModalCreateUser.tsx [hunk 2]
```diff
@@ -168,17 +170,21 @@ const ModalForm: React.FC<Props> = ({ form, user, groups, viewOnly, roles }) =>
...
+                      disabled={
+                        roles?.find((ro) => ro.id === r.id)?.fromGroup?.length ||
+                        roles?.find((ro) => ro.id === r.id)?.fromWorkspace?.length
+                      }
+                      key={r.id}
+                      value={r.id}>
+                      {r.name}
+                    </Select.Option>
+                  )),
+                NotLoaded: () => undefined,
```
Comment (keita-determined) [inline webui/react/src/hooks/useModal/UserSettings/useModalCreateUser.tsx:186]: same here

https://github.com/determined-ai/determined/pull/5522#discussion_r1042317154

We should add loading state (spinner) or `// TODO show spinner when this is loading`

**Review by keita-determined — COMMENTED** [review]
i wonder why test fails

File: webui/react/src/hooks/useModal/UserSettings/useModalCreateUser.tsx (hunk 2 above)
Comment (gt2345) [reply webui/react/src/hooks/useModal/UserSettings/useModalCreateUser.tsx]: @keita-determined  Something might happened to Github at that time, it's working now

File: webui/react/src/stores/knowRoles.tsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+import { UserRole } from 'types';
+import handleError from 'utils/error';
+import { Loadable, Loaded, NotLoaded } from 'utils/loadable';
+
+type KnownRolesContext = {
+  knownRoles: Loadable<UserRole[]>;
+  updateKnowRoles: React.Dispatch<React.SetStateAction<Loadable<UserRole[]>>>;
+};
+
+export const initKnowRoles: UserRole[] = [];
```
Comment (keita-determined) [inline webui/react/src/stores/knowRoles.tsx:14]: i dont think we dont need it

File: webui/react/src/hooks/useModal/UserSettings/useModalCreateGroup.tsx [hunk 4]
```diff
@@ -47,7 +48,8 @@ const ModalForm: React.FC<Props> = ({ form, users, group, groupRoles }) => {
   const { canModifyPermissions } = usePermissions();
   const [isLoading, setIsLoading] = useState(true);
 
-  const { knownRoles } = useStore();
+  const knowRolesLoadable = useKnownRoles();
+  const knownRoles = Loadable.getOrElse(initKnowRoles, knowRolesLoadable);
```
Comment (keita-determined) [inline webui/react/src/hooks/useModal/UserSettings/useModalCreateGroup.tsx:52]: `initKnowRoles` should be just `[]`

**Review by keita-determined — APPROVED** [review]
LGTM

**Comment by keita-determined** [comment]
missing ticket number btw

**Pull request merged**